### PR TITLE
Barcode login

### PR DESCRIPTION
**Pull request by tampakis**
- Adds omniauth "strategy" for barcode login. Allows users without a valid net id (guest patrons) to login to check their account.
 - Relaxed the requirement for emails to be unique (in theory we can now let users set a preferred email address).
 - Users logging out from CAS get redirected to Princeton CAS logout page, but Barcode users get redirected to catalog home page
 - Users with a valid net id that try logging in via barcode get redirected to cas instead

### CODE REVIEW

File: config/initializers/session_store.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.session_store :cookie_store, key: '_orangelight_session',
-                                                      expire_after: 30.minutes
+# Rails.application.config.session_store :cookie_store, key: '_orangelight_session',
+#                                                       expire_after: 30.minutes
```
Comment (eliotjordan) [inline config/initializers/session_store.rb]: Is there a reason to keep this file?

Comment (tampakis) [inline config/initializers/session_store.rb]: It's there as a reminder for https://github.com/pulibrary/orangelight/issues/585

File: db/schema.rb [hunk 2]
```diff
@@ -66,6 +66,15 @@
...
 
+  create_table "orangelight_name_titles", force: :cascade do |t|
+    t.text    "label"
+    t.integer "count"
+    t.text    "sort"
+    t.string  "dir"
+  end
+
+  add_index "orangelight_name_titles", ["sort"], name: "index_orangelight_name_titles_on_sort", using: :btree
+
```
Comment (eliotjordan) [inline db/schema.rb]: Is this table part of the barcode / cas login? If not, perhaps it should move to a different PR.

Comment (tampakis) [inline db/schema.rb]: Oh good catch!

**Comment by eliotjordan** [comment]
Nice work!  I really like the concise way you write some of the specs. Especially the allow statements. I imagine this will go into pulmap/lime/turnip sometime soon. A few things:
  1. How about increasing the [bibdata.rb](https://coveralls.io/builds/7430293/source?filename=lib%2Forangelight%2Fbibdata.rb#L3) test coverage to 100 by adding a `500` status test and a `ConnectionFailed` test to the [account controller spec](https://github.com/pulibrary/orangelight/blob/development/spec/controllers/account_controller_spec.rb#L45)? It seems like as a good time as any since you pulled out logic to create this new module.
  1. It would be nice if the barcode tooltips had more specific messages. E.g. `Please fill in your last name` and `Please enter a valid barcode`. Not a deal breaker, but it would be good to at least make an issue for it.

File: config/initializers/session_store.rb (hunk 1 above)
Comment (eliotjordan) [inline config/initializers/session_store.rb]: Put the snippet in the issue as a reminder instead?

**Comment by tampakis** [comment]
@eliotjordan - addressed the coverage for the get_patron method and opened #663.

**Comment by eliotjordan** [comment]
👍
